---
layout: post
title: Selected Projects
---

## Tools for Analyzing Next Generation Sequence Data

### *systemPipeR*: NGS workflow and report generation environment

[*systemPipeR*](http://bioconductor.org/packages/devel/systemPipeR/){:target="_blank"} is an
R/Bioconductor package for building and running automated analysis workflows
for a wide range of next generation sequence (NGS) applications. Important
features include a uniform workflow interface across different NGS
applications, automated report generation, and support for running both R and
command-line software, such as NGS aligners or peak/variant callers, on local
computers or compute clusters. Efficient handling of complex sample sets and
experimental designs is facilitated by a consistently implemented sample
annotation infrastructure. 

![systempiper](/public/images/systempiper.png){: .center-image }
<font size="3"><b>Figure 1:</b> Workflow design structure of <i>systemPipeR</i>.</font>

<br/>

### Reference-Assisted Transcriptome Assembly

Owing to the complexity and often incomplete representation of transcripts in
RNA-Seq libraries, the assembly of high-quality transcriptomes can be extremely
challenging. To improve this, my group is developing 
algorithms for guiding these assemblies with genomic sequences of related organisms as
well as reducing the complexity in NGS libraries. The software tools we have published for this
purpose so far include *SEED* [(Bao et al., 2011)](http://www.hubmed.org/display.cgi?uids=21810899)
and *BRANCH* [(Bao et al., 2013)](http://www.hubmed.org/display.cgi?uids=23493323). *BRANCH*
is a reference assisted post-processing method for enhancing *de novo* 
transcriptome assemblies (Figure 2). It can be used in combination with most de novo
transcriptome assembly software tools. The assembly improvements are achieved
with help from partial or complete genomic sequence information. They can be
obtained by sequencing and assembling a genomic DNA sample in addition to the
RNA samples required for a transcriptome assembly project. This approach is
practical because it requires only preliminary genome assembly results in form
of contigs. Nowadays, the latter can be generated with very reasonable cost and
time investments. In case the genome sequence of a closely related organism is
available, one can skip the genome assembly step and use the related gene
sequences instead. This type of reference assisted assembly approach provides
many attractive opportunities for improving de novo NGS assemblies in the
future by making use of the rapidly growing number of reference genome
information available to us.

<br/>

![BRANCH Image](/public/images/branch.jpg)
<font size="3"><b>Figure 2:</b> Outline of *BRANCH* algorithm published in <a href="http://www.hubmed.org/display.cgi?uids=23493323">Bao et al. 2013</a>. (a) Read alignments against preassembled transcripts  and closely related genomic reference. (b) Junction graph based on this alignment result. (c) Assembly of extended transcripts.</font>

<br/>

### Modeling Gene Expression Networks from RNA-Seq and ChIP-Seq Data

As part of several collaborative research projects, my group has developed a
variety of data analysis pipelines for profiling data from next generation
sequencing projects (e.g. RNA-Seq and ChIP-Seq), microarray experiments and
high-throughput small molecule screens. Most of the data analysis resources
developed by these projects are described in the associated online manuals for
[next generation data analysis](http://manuals.bioinformatics.ucr.edu/home).
Recent research publications of these projects include: 
[Yang et al., 2013](http://www.hubmed.org/display.cgi?uids=23793751){:target="_blank"}; 
[Zou et al., 2013](http://www.hubmed.org/display.cgi?uids=23633570){:target="_blank"}; 
[Yadav et al., 2013](http://www.hubmed.org/display.cgi?uids=23633570){:target="_blank"}; 
[Yadav et al., 2011](http://www.hubmed.org/display.cgi?uids=21979915){:target="_blank"}; 
[Mustroph et al., 2009](http://www.hubmed.org/display.cgi?uids=19843695){:target="_blank"}. 

<br/>

## Software Resources for Small Molecule Discovery and Chemical Genomics

Software tools for modeling the similarities among drug-like small molecules
and high-throughput screening data are important for many applications in drug
discovery and chemical genomics. In this area we are working on the development
of the [ChemmineR](http://manuals.bioinformatics.ucr.edu/home/chemminer){:target="_blank"}
environment ([Cao et al., 2008](http://www.hubmed.org/display.cgi?uids=18596077){:target="_blank"}; 
[Backman et al., 2011](http://www.hubmed.org/display.cgi?uids=21576229){:target="_blank"}). This modular
software infrastructure consists currently of five R/Bioconductor packages along with a
user-friendly web interface, named *ChemMine Tools* that
is intended for non-expert users (Figures 3-4). The integration of cheminformatic 
tools with the R programming environment has many advantages for small molecule discovery, such as easy access to a wide spectrum
of statistical methods, machine learning algorithms and graphic utilities.
Currently, the ChemmineR toolkit
provides utilities for processing large numbers of molecules,
physicochemical/structural property predictions, structural similarity
searching, classification and clustering of compound libraries and screening
results with a wide spectrum of algorithms. More recently, we have developed
for this infrastructure the *fmcsR* algorithm which is the first mismatch tolerant 
maximum common substructure search tool in
the field ([Wang et al., 2013](http://www.hubmed.org/display.cgi?uids=23962615){:target="_blank"}).
In our comparisons with related structure similarity search tools, *fmcsR*
showed the best virtual screening (VS) performance.

<br/>

![chemm](/public/images/ChemmineR.png){: .center-image }

![fig1](/public/images/fig1b.png){: .center-image }

<font size="3"><b>Figure 3:</b> <i>ChemmineR</i> small molecule modeling environment with its add-on packages and selected functionalities.</font>

<br/>

![crosstarget](/public/images/crosstarget.png){: .center-image }


<font size="3"><b>Figure 4:</b> Selectivity Analysis with <i>ChemmineR</i> and <i>bioassayR</i></font>

<br/>

## Functional Annotation of Gene and Protein Sequences

Computational methods for characterizing the functions of protein sequences
play an important role in the discovery of novel molecular, biochemical and
regulatory activities. To facilitate this process, we have developed the
sub-HMM algorithm that extends the application spectrum of profile HMMs to
motif discovery and active site prediction in protein sequences ([Horan et al.
2010](http://www.hubmed.org/display.cgi?uids=20420695){:target="_blank"}). Its most interesting
utility is the identification of the functionally relevant residues in proteins
of known and unknown function (Figure 5). Additionally, sub-HMMs can be used
for highly localized sequence similarity searches that focus on shorter
conserved features rather than entire domains or global similarities. As part
of this study we have predicted a comprehensive set of putative active sites
for all protein families available in the Pfam database which has become a
valuable knowledge resource for characterizing protein functions in the future.

![hmmlogo](/public/images/hmmlogo.jpg){: .center-image }


<font size="3"><b>Figure 5:</b> Illustration of the sub-HMM extraction process from conserved protein domains, here Pfam desaturase domain (PF00487).</font>

<br/>

## References

* see [Publication List]({{site.baseurl}}/pubs/)



