---
layout: post
title: Selected Projects
---

## Tools for Analyzing Next Generation Sequence Data

### Reference-Assisted Transcriptome Assembly

Owing to the complexity and often incomplete representation of transcripts in
RNA-Seq libraries, the assembly of high-quality transcriptomes can be extremely
challenging. To improve this, my group is developing 
algorithms for guiding these assemblies with genomic sequences of related organisms as
well as reducing the complexity in NGS libraries. The software tools we have published for this
purpose so far include *SEED* [(Bao et al., 2011)](http://www.hubmed.org/display.cgi?uids=21810899)
and *BRANCH* [(Bao et al., 2013)](http://www.hubmed.org/display.cgi?uids=23493323). BRANCH
is a reference assisted post-processing method for enhancing *de novo* 
transcriptome assemblies (Figure 1). It can be used in combination with most de novo
transcriptome assembly software tools. The assembly improvements are achieved
with help from partial or complete genomic sequence information. They can be
obtained by sequencing and assembling a genomic DNA sample in addition to the
RNA samples required for a transcriptome assembly project. This approach is
practical because it requires only preliminary genome assembly results in form
of contigs. Nowadays, the latter can be generated with very reasonable cost and
time investments. In case the genome sequence of a closely related organism is
available, one can skip the genome assembly step and use the related gene
sequences instead. This type of reference assisted assembly approach provides
many attractive opportunities for improving de novo NGS assemblies in the
future by making use of the rapidly growing number of reference genome
information available to us.

<br/>

![BRANCH Image](/public/images/branch.jpg)
<font size="3"><b>Figure 1:</b> Outline of BRANCH algorithm published in <a href="http://www.hubmed.org/display.cgi?uids=23493323">Bao et al. 2013</a>. (a) Read alignments against preassembled transcripts  and closely related genomic reference. (b) Junction graph based on this alignment result. (c) Assembly of extended transcripts.</font>

<br/>

### Modeling Gene Expression Networks from RNA-Seq and ChIP-Seq Data

As part of several collaborative research projects, my group has developed a
variety of data analysis pipelines for profiling data from next generation
sequencing projects (e.g. RNA-Seq and ChIP-Seq), microarray experiments and
high-throughput small molecule screens. Most of the data analysis resources
developed by these projects are described in the associated online manuals for
[next generation data analysis](http://manuals.bioinformatics.ucr.edu/home).
Recent research publications of these projects include: 
[Yang et al., 2013](http://www.hubmed.org/display.cgi?uids=23793751); 
[Zou et al., 2013](http://www.hubmed.org/display.cgi?uids=23633570); 
[Yadav et al., 2013](http://www.hubmed.org/display.cgi?uids=23633570); 
[Yadav et al., 2011](http://www.hubmed.org/display.cgi?uids=21979915); 
[Mustroph et al., 2009](http://www.hubmed.org/display.cgi?uids=19843695). 

<br/>

## Software Resources for Small Molecule Discovery and Chemical Genomics

Software tools for modeling the similarities among drug-like small molecules
and high-throughput screening data are important for many applications in drug
discovery and chemical genomics. In this area we are working on the development
of the [ChemmineR](http://manuals.bioinformatics.ucr.edu/home/chemminer)
environment ([Cao et al., 2008](http://www.hubmed.org/display.cgi?uids=18596077); 
[Backman et al., 2011](http://www.hubmed.org/display.cgi?uids=21576229)). This modular
software infrastructure consists currently of five R/Bioconductor packages along with a
user-friendly web interface, named *ChemMine Tools* that
is intended for non-expert users (Figures 2-3). The integration of cheminformatic 
tools with the R programming environment has many advantages for small molecule discovery, such as easy access to a wide spectrum
of statistical methods, machine learning algorithms and graphic utilities.
Currently, the ChemmineR toolkit
provides utilities for processing large numbers of molecules,
physicochemical/structural property predictions, structural similarity
searching, classification and clustering of compound libraries and screening
results with a wide spectrum of algorithms. More recently, we have developed
for this infrastructure the *fmcsR* algorithm which is the first mismatch tolerant 
maximum common substructure search tool in
the field ([Wang et al., 2013](http://www.hubmed.org/display.cgi?uids=23962615)).
In our comparisons with related structure similarity search tools, *fmcsR*
showed the best virtual screening (VS) performance.

<br/>

![chemm](/public/images/ChemmineR.png){: .center-image }

![fig1](/public/images/fig1b.png){: .center-image }

<font size="3"><b>Figure 2:</b> <i>ChemmineR</i> small molecule modeling environment with its add-on packages and selected functionalities.</font>

<br/>

![crosstarget](/public/images/crosstarget.png){: .center-image }


<font size="3"><b>Figure 3:</b> Selectivity Analysis with <i>ChemmineR</i> and <i>bioassayR</i></font>

<br/>

## Functional Annotation of Gene and Protein Sequences

<div>Computational methods for characterizing the functions of protein sequences play an important role in the discovery of novel molecular, biochemical and regulatory activities. To facilitate this process, we have developed the sub-HMM algorithm that <span style="background-color:transparent;line-height:1.6;font-size:10pt">extends the application spectrum of profile HMMs to motif discovery and active site prediction in protein sequences ([Horan et al. 2010](http://www.hubmed.org/display.cgi?uids=20420695)). Its most interesting utility is the identification of the functionally relevant residues in proteins of known and unknown function (Figure 4). Additionally, sub-HMMs can be used for highly localized sequence similarity searches that focus on shorter conserved features rather than entire domains or global similarities. As part of this study we have predicted </span><span style="background-color:transparent;line-height:1.6;font-size:10pt">a comprehensive set of putative active sites for all protein families available in the Pfam database which has become</span><span style="background-color:transparent;font-size:10pt;line-height:1.6"> a valuable knowledge resource for characterizing protein functions in the future.</span></div>

<div>![](https://sites.google.com/a/bioinformatics.ucr.edu/girkelab/home/logo.jpg)</div>

<div>

<div style="text-align:center">**Figure 4:** Illustration of the sub-HMM extraction process from conserved protein domains, here Pfam <span style="background-color:transparent">desaturase domain (PF00487).</span></div>

<div style="text-align:center">.</div>

<span style="color:rgb(68,68,68);font-size:1.6em;background-color:transparent;line-height:1.6">References</span></div>

<div>

*   see [Publication List](https://sites.google.com/a/bioinformatics.ucr.edu/girkelab/home/publications)
*   [Google Scholar Profile](http://scholar.google.com/citations?user=QjoDLv4AAAAJ&hl=en&oi=ao)

<div>

<div><a name="position"></a>

## Open Positions

_Postdoctoral position in comparative genomics - <font color="#ff0000">filled</font>_  

<div style="margin-left:40px">This position requires several years of experience in computational biology, statistics, database design and data mining with strong publications records in several of these areas. The candidate should be proficient in at least one of the common programming languages that are used in bioinformatics: C, Python, Java, Perl or R. Experience with web and database programming is also beneficial. To apply for this position, please email your CV and a statement of research interests to thomas.girke@ucr.edu.  
</div>

_Postdoctoral position in cheminformatics - __<font color="#ff0000">filled</font>_  

<div style="margin-left:40px">This position requires experience in computational chemistry, drug-informatics, QSAR/phramacophore modeling and data mining with strong publications records in several of these areas. The candidate should be proficient in at least one of the common programming languages that are used in cheminformatics: C, Python, Java, Perl or R. Experience with web and database programming is also beneficial. To apply for this position, please email your CV and a statement of research interests to thomas.girke@ucr.edu.  
</div>

_Undergraduate research positions __- __<font color="#ff0000">filled</font>_  

<div style="margin-left:40px">Several postions are available for undergraduate students who are interested in participating in challenging database and software development projects for various research projects in bioinformatics and cheminformatics. The required skills are experience with open-source databases (MySQL/PostgreSQL), application and web programming with Python, Perl and/or Java. The minimum time commitment is 15-20h per week for a period of at least 18 months. Full-time employment during the summer break is possible. To apply for these positions, please email your CV with a detailed outline of your computational skills to thomas.girke@ucr.edu.  
</div>

<a name="labmeeting"></a>  
</div>

</div>

</div>
